### ISSUE TITLE
[QUESTION] Is it possible to use ImageOptimize with CraftQL?

### ISSUE THREAD

**Issue by stursby** [nystudio107/craft3-imageoptimize#71]
I've got ImageOptimize setup correctly (w/ Imgix) and I'm trying to see if it works out of the box with [CraftQL](https://github.com/markhuot/craftql).

Here's my sample query

![image](https://user-images.githubusercontent.com/527849/38265070-6cd8abe0-373a-11e8-982f-54ea8020219a.png)

And I see, what appears to be, the default Craft transforms

![image](https://user-images.githubusercontent.com/527849/38265160-af260948-373a-11e8-8605-99abcc1464e1.png)

But wondering if I can get back the Imgix srcet?

Any help would be much appreciated!!

**Comment by khalwat** [comment]
I’m pretty sure GraphQL would need to know about the ImageOptmize field one way or another. I’ll see about adding that support on my end

**Comment by khalwat** [comment]
@markhuot the author of CraftQL is making a few changes to the docs and APIs for third party integrations like this; once that's done, I'm planning to have a look!

**Comment by khalwat** [comment]
Progress!

```
{
  entries(section:[homepage], limit:5) {
    ...on Homepage {
      title
      url
      someAsset {
        ...on AssetsVolume {
        title
        optimizedImages {
          ...on OptimizedImagesData {
            optimizedImageUrls
          }
          
        }
        }
      }
    }
  }
}
```

Results in:
```
{
  "data": {
    "entries": [
      {
        "title": "Homepage",
        "url": "http://craft3.test/",
        "someAsset": [
          {
            "title": "Maxresdefault",
            "optimizedImages": {
              "optimizedImageUrls": [
                "http://craft3.test/assets/_1170x658_crop_center-center_82_line/maxresdefault.jpg",
                "http://craft3.test/assets/_970x545_crop_center-center_82_line/maxresdefault.jpg",
                "http://craft3.test/assets/_750x562_crop_center-center_60_line/maxresdefault.jpg",
                "http://craft3.test/assets/_320x240_crop_center-center_60_line/maxresdefault.jpg"
              ]
            }
          }
        ]
      }
    ]
  }
}
```

**Comment by khalwat** [comment]
Implemented in 1.4.31:

https://github.com/nystudio107/craft-imageoptimize/releases/tag/1.4.31

https://github.com/nystudio107/craft-imageoptimize#graphql-via-craftql-plugin

**Comment by jpcarpenter** [comment]
Is there an equivalent implementation in doing this with the newly integrated GraphQL into Craft? Me and my colleague have been scratching our heads on this one for a while.
